### PR TITLE
Separate SQL functions

### PR DESCRIPTION
**Pull request by Changaco**
Implements #3162.

### CODE REVIEW

**Comment by whit537** [comment]
Maybe move `branch.sql` into `sql/` as well?

File: sql/upsert_community.sql [hunk 1]
```diff
@@ -0,0 +1,46 @@
+CREATE OR REPLACE FUNCTION upsert_community() RETURNS trigger AS $$
```
Comment (whit537) [inline sql/upsert_community.sql]: Why do we use `CREATE OR REPLACE` here but just `CREATE` for `enumerate`, above? It seems that we should have one pattern. I think I would now expect `CREATE OR REPLACE` for functions.

**Comment by Changaco** [comment]
Anything else ?

**Comment by whit537** [comment]
Yeah, I don't understand how this interacts with whatever we're doing now to keep dev and prod in sync.

**Comment by whit537** [comment]
I also don't understand how this affects deployment. If I want to modify `upsert_community`, do I still use branch.sql? If not, how do my changes get deployed?

**Comment by Changaco** [comment]
> I don't understand how this interacts with whatever we're doing now to keep dev and prod in sync.

`make schema-diff` doesn't care about what file(s) the schema is in. I've run it from this branch, it doesn't detect any drift.

> If I want to modify `upsert_community`, do I still use branch.sql?

Yes, you make a `branch.sql` with an `\i sql/upsert_community.sql` inside.

**Comment by whit537** [comment]
> I've run it from this branch and it doesn't detect any drift.

I get the following error from `make schema-diff`:

```shell
[...]
pg_dump -sO `heroku config:get DATABASE_URL -a gratipay` >prod.sql
env/bin/honcho run -e defaults.env,tests/test.env,tests/local.env sh -c 'pg_dump -sO "$DATABASE_URL"' >local.sql
diff -uw prod.sql local.sql
--- prod.sql    2015-03-02 09:08:05.000000000 -0500
+++ local.sql   2015-03-02 09:08:06.000000000 -0500
@@ -24,20 +24,6 @@
 
 
 --
--- Name: pg_stat_statements; Type: EXTENSION; Schema: -; Owner: -
---
-
-CREATE EXTENSION IF NOT EXISTS pg_stat_statements WITH SCHEMA public;
-
-
---
--- Name: EXTENSION pg_stat_statements; Type: COMMENT; Schema: -; Owner: -
---
-
-COMMENT ON EXTENSION pg_stat_statements IS 'track execution statistics of all SQL statements executed';
-
-
---
 -- Name: pg_trgm; Type: EXTENSION; Schema: -; Owner: -
 --
 
make: *** [schema-diff] Error 1
$
```

**Comment by Changaco** [comment]
[`schema.sql` does execute these two commands](https://github.com/gratipay/gratipay.com/blob/separate-sql-functions/sql/schema.sql#L12-13). Maybe they're failing on your system for some reason, do you see an error message ?

**Comment by whit537** [comment]
Indeed:

```
ERROR:  could not open extension control file "/usr/local/pgsql/share/extension/pg_stat_statements.control": No such file or directory
ERROR:  extension "pg_stat_statements" does not exist
```

**Comment by whit537** [comment]
I built and installed the extension and now `make schema-diff` exits without error.

**Pull request merged**